tests/backup/backup_kubevirt_test.go

```diff
@@ -1760,3 +1761,549 @@ var _ = Describe("{KubevirtInPlaceRestoreWithReplaceAndRetain}", func() {
+		scheduleNames              []string
+		scheduledAppContexts       []*scheduler.Context
+		singleScheduledAppContexts []*scheduler.Context
+		multiScheduledAppContexts  []*scheduler.Context
+		sourceClusterUID           string
+		cloudCredName              string
+		cloudCredUID               string
+		backupLocationUID          string
+		backupLocationName         string
+		backupLocationMap          map[string]string
+		labelSelectors             map[string]string
+		namespaceMapping           map[string]string
+		providers                  []string
+		numOfDeployments           int
+		appNamespaces              []string
+		schPolicyUid               string
+		periodicPolicyName         string
+		preRuleName                string
+		postRuleName               string
+		preRuleUid                 string
+		postRuleUid                string
+		preRuleNames               []string
+		postRuleNames              []string
+		allVMs                     []kubevirtv1.VirtualMachine
+		allVMNames                 []string
+		preRuleList                []*api.RulesInfo_RuleItem
+		postRuleList               []*api.RulesInfo_RuleItem
+		backupNameToAppContextMap  map[string][]*scheduler.Context
+		restoreNameBackupNameMap   map[string]string
+	)
+
+	backupLocationMap = make(map[string]string)
+	backupNames = make([]string, 0)
+	restoreNames = make([]string, 0)
+	replaceRestoreNames = make([]string, 0)
+	labelSelectors = make(map[string]string)
+	appNamespaces = make([]string, 0)
+	backupNamespaceMap := make(map[string]string)
+	restoreNameBackupNameMap = make(map[string]string)
+	backupNameToAppContextMap = make(map[string][]*scheduler.Context)
+	timeStamp := strconv.Itoa(int(time.Now().Unix()))
+	periodicPolicyName = fmt.Sprintf("%s-%s", "periodic", timeStamp)
+	backupDriver := Inst().Backup
+
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("DefaultBackupRestoreWithKubevirtAndNonKubevirtNS", "Verify default backup & restore with both Kubevirt and Non-Kubevirt namespaces", nil, 93006, Vpinisetti, Q1FY25)
+		numOfDeployments = Inst().GlobalScaleFactor
+		providers = GetBackupProviders()
+		log.InfoD("Deploying all provided applications in a single namespace")
+		for i := 0; i < numOfDeployments; i++ {
+			taskName := fmt.Sprintf("%s-%d-%d", TaskNamePrefix, 93006, i)
+			namespace := fmt.Sprintf("single-ns-multi-app-%s-%v", taskName, time.Now().Unix())
+			appContexts := ScheduleApplicationsOnNamespace(namespace, taskName)
+			appNamespaces = append(appNamespaces, namespace)
+			for index, appCtx := range appContexts {
+				appName := Inst().AppList[index]
+				appCtx.ReadinessTimeout = AppReadinessTimeout
+				log.InfoD("Scheduled applications [%s] in source cluster in single namespace [%s]", appName, namespace)
+				singleScheduledAppContexts = append(singleScheduledAppContexts, appCtx)
+				log.InfoD("Single Scheduled App Contexts %v : %v", i, singleScheduledAppContexts)
+				scheduledAppContexts = append(scheduledAppContexts, appCtx)
+			}
```

Comment:
Why are you looping over `appContexts` here? You can directly collect the list from `ScheduleApplicationsOnNamespace` like this
```
singleScheduledAppContexts := ScheduleApplicationsOnNamespace(namespace, taskName)
```

Reply:
done